client/components/header.jsx

```diff
@@ -3,9 +3,15 @@ import React from 'react';
 class Header extends React.Component {
 
   render() {
+    let sum = 0;
+    for (let i = 0; i < this.props.grades.length; i++) {
+      sum += this.props.grades[i].grade;
+    }
+    const average = sum / this.props.grades.length;
```

Comment:
You should do this math inside a method in `App` named `getAverageGrade()`. Then in the `render()` method of `App`, call that method `getAverageGrade()` and pass the return value into the `Header`.